File: scripts/test_solutions/056_merge_intervals.py
def merge(intervals):
    if not intervals:
        return []
    # Handle nested list input: [[[1,3],[2,6]]] -> [[1,3],[2,6]]
    if intervals and isinstance(intervals[0], list) and intervals[0] and isinstance(intervals[0][0], list):
        intervals = intervals[0]
    intervals.sort(key=lambda x: x[0])
    result = [intervals[0][:]]
    for i in range(1, len(intervals)):
        last = result[-1]
        if intervals[i][0] <= last[1]:
            last[1] = max(last[1], intervals[i][1])
        else:
            result.append(intervals[i][:])
    return result

File: scripts/test_solutions/056_merge_intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        if (intervals.empty()) {
            return {};
        }
        sort(intervals.begin(), intervals.end(), [](const vector<int>& a, const vector<int>& b) {
            return a[0] < b[0];
        });
        vector<vector<int>> result;
        result.push_back(intervals[0]);
        for (int i = 1; i < intervals.size(); i++) {
            vector<int>& last = result.back();
            if (intervals[i][0] <= last[1]) {
                last[1] = max(last[1], intervals[i][1]);
            } else {
                result.push_back(intervals[i]);
            }
        }
        return result;
    }
};
